src/main/java/bio/terra/service/resourcemanagement/BufferService.java

```diff
@@ -0,0 +1,131 @@
+package bio.terra.service.resourcemanagement;
+
+import bio.terra.buffer.api.BufferApi;
+import bio.terra.buffer.api.UnauthenticatedApi;
+import bio.terra.app.configuration.ResourceBufferServiceConfiguration;
+import bio.terra.buffer.client.ApiClient;
+import bio.terra.buffer.client.ApiException;
+import bio.terra.buffer.model.HandoutRequestBody;
+import bio.terra.buffer.model.PoolInfo;
+import bio.terra.buffer.model.ResourceInfo;
+import bio.terra.buffer.model.SystemStatus;
+import bio.terra.buffer.model.SystemStatusSystems;
+import bio.terra.model.RepositoryStatusModelSystems;
+import bio.terra.service.resourcemanagement.exception.BufferServiceAPIException;
+import bio.terra.service.resourcemanagement.exception.BufferServiceAuthorizationException;
+import java.io.IOException;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+
+/** A service for integrating with the Resource Buffer Service. */
+@Component
+public class BufferService {
+    private final Logger logger = LoggerFactory.getLogger(BufferService.class);
+
+    private final ResourceBufferServiceConfiguration bufferServiceConfiguration;
+
+    @Autowired
+    public BufferService(ResourceBufferServiceConfiguration bufferServiceConfiguration) {
+        this.bufferServiceConfiguration = bufferServiceConfiguration;
+    }
+
+    private ApiClient getApiClient(String accessToken) {
+        ApiClient client = new ApiClient();
+        client.setAccessToken(accessToken);
+        client.setBasePath(bufferServiceConfiguration.getInstanceUrl());
+        return client;
+    }
+
+    private ApiClient getUnAuthApiClient() {
```

Comment:
What's `UnAuth`?

Reply:
The `status` endpoint is unauthenticated: https://buffer.dsde-dev.broadinstitute.org/swagger-ui.html#/